### PR TITLE
Line Message API (Independent Publisher)

### PR DESCRIPTION
**Pull request by felaray**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md. 

image
image
image

`paconn validate`
image

### CODE REVIEW

**Comment by Amjed-Ayoub** [comment]
Hello @felaray,

Thank you so much for submitting this new proposal. I can see that you have submitted the connector artifacts, if you would like us to review and publish this connector, please remove the proposal from the PR title. Thank you.

**Comment by felaray** [comment]
I was removed "proposal" from the title. thanks!

File: independent-publisher-connectors/Line Message/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                    }
+                  },
+                  "description": "messages"
+                }
+              }
+            }
+          }
+        ],
+        "responses": {
+          "default": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Line Message/apiDefinition.swagger.json]: @felaray it is not allowed to have a default response only. Please define a proper response. please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/coding-standards#responses).

File: independent-publisher-connectors/Line Message/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            "description": "default",
+            "schema": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Line Message/apiDefinition.swagger.json]: @felaray empty schema is not allowed, please define a proper schema.

File: independent-publisher-connectors/Line Message/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  ],
+  "tags": [],
+  "x-ms-connector-metadata": [
+    {
+      "propertyName": "Website",
+      "propertyValue": "https://developers.line.biz/en/services/messaging-api/"
+    },
+    {
+      "propertyName": "Privacy policy",
+      "propertyValue": "https://developers.line.biz/en/services/messaging-api/"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Line Message/apiDefinition.swagger.json]: @felaray can you please provide a valid URL for the privacy policy page? Maybe "https://developers.line.biz/en/terms-and-policies/"

File: independent-publisher-connectors/Line Message/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        "description": "Send a message to the associated user ID.",
+        "operationId": "SendMessage",
+        "parameters": [
+          {
+            "name": "Content-Type",
+            "in": "header",
+            "required": true,
+            "type": "string",
+            "default": "application/json",
+            "x-ms-visibility": "internal"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Line Message/apiDefinition.swagger.json:35]: @felaray can you please clarify, why this has been marked as internal parameter?

File: independent-publisher-connectors/Line Message/apiProperties.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+  "properties": {
+    "connectionParameters": {},
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Line Message/apiProperties.json]: @felaray In the swagger file you mentioned that the connector uses API Key, May I ask you please to setup the "connectionParameters" correctly? Please see an example [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/independent-publisher-connectors/Leap-Ai/apiProperties.json).

File: independent-publisher-connectors/Line Message/apiProperties.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+    "iconBrandColor": "#00ff00",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Line Message/apiProperties.json]: @felaray for all our Independent Publishers we ask to set the "iconBrandColor" to "#da3b01". Please see our template [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiProperties.json).

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Comment by felaray** [comment]
@Amjed-Ayoub Thanks for reply,I'll update the swagger file based on your suggestion.

File: independent-publisher-connectors/Line Message/apiDefinition.swagger.json (hunk 4 above)
Comment (felaray) [reply independent-publisher-connectors/Line Message/apiDefinition.swagger.json:35]: @Amjed-Ayoub I believe that users do not need to modify this parameter. If mark it as an internal parameter, users will not accidentally modify it.

File: independent-publisher-connectors/Line Message/apiDefinition.swagger.json (hunk 1 above)
Comment (felaray) [reply independent-publisher-connectors/Line Message/apiDefinition.swagger.json]: @Amjed-Ayoub I have already added the expected response. (200/400/401)

File: independent-publisher-connectors/Line Message/apiDefinition.swagger.json (hunk 2 above)
Comment (felaray) [reply independent-publisher-connectors/Line Message/apiDefinition.swagger.json]: @Amjed-Ayoub Is the "schema" field required? The schema field doesn't appear to be required, so I have removed it.

File: independent-publisher-connectors/Line Message/apiDefinition.swagger.json (hunk 3 above)
Comment (felaray) [reply independent-publisher-connectors/Line Message/apiDefinition.swagger.json]: @Amjed-Ayoub I was renew the url.

File: independent-publisher-connectors/Line Message/apiProperties.json (hunk 6 above)
Comment (felaray) [reply independent-publisher-connectors/Line Message/apiProperties.json]: @Amjed-Ayoub Done this.

File: independent-publisher-connectors/Line Message/apiProperties.json (hunk 5 above)
Comment (felaray) [reply independent-publisher-connectors/Line Message/apiProperties.json]: @Amjed-Ayoub Thanks for example.I have already update connectionParameters.

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @felaray ,
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by Amjed-Ayoub** [comment]
Hello @felaray ,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

**Review by Amjed-Ayoub — APPROVED** [review]
Hello @felaray ,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.

**Pull request merged**